tcms/testcases/forms.py

```diff
@@ -6,6 +6,7 @@
 from tcms.core.forms.fields import UserField
 from tcms.core.utils import string_to_list
 from tcms.core.widgets import SimpleMDE
+from durationwidget.widgets import TimeDurationWidget
```

Comment:
Missing module `durationwidget`. Where is this coming from ?

Reply:
I imported the duration widget by following [this doc](https://pypi.org/project/django-durationwidget/) and added it into the settings `INSTALLED_APPS = [
    ...
    'durationwidget',
]`

Reply:
> ... and added it into the settings `INSTALLED_APPS = [ ... 'durationwidget', ]`

You definitely didn't because the pull request is missing this change!

Reply:
Another thing that you do is randomly mixing & matching 2 different approaches to implement the requested feature which isn't going to work. 

`django-durationwidget` is a Python package which seems to implement a widget similar to what we need. Whether or not we can use that particular implementation remains to be seen. I wasn't aware of this package before. If you would like to follow this approach then implement what their README says so we can see how the widget looks like when rendered on the page and figure out if there are any other issues or not. 

`bootstrap-duration-picker` is a Node.js package which provides the JavaScript & CSS implementation of what we need. It is built on top of bootstrap and will fit in more easily (probably) with the existing UI b/c the Patternfly library is also built on top of bootstrap.  This package is missing the Python code which allows you to easily glue the backend and frontend pieces together. Parts of this PR are trying to implement this missing glue layer but there is more work that needs to be done. 

Pick one of the approaches and go with it but don't mix half-and-half of both. That mix will never going to work!